pkg/mpc-examples/contracts/03-ebu-rebalancer/EBURebalancerController.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/ILastCreatedPoolFactory.sol";
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/IManagedPool.sol";
+import "@balancer-labs/v2-interfaces/contracts/vault/IVault.sol";
+import "@balancer-labs/v2-pool-utils/contracts/lib/ComposablePoolLib.sol";
+
+// solhint-disable not-rely-on-time
+
+contract EbuRebalancerController {
+    IVault private immutable _vault;
+    bytes32 private immutable _poolId;
+    IERC20[] private _tokens;
+
+    uint256 private constant _MINIMUM_DURATION_BETWEEN_REBALANCE = 30 days;
+    uint256 private constant _REBALANCE_DURATION = 7 days;
+    uint256 private constant _MIN_PAUSE_DURATION = 7 days;
+
+    uint256 private constant _MAX_SWAP_FEE_PERCENTAGE = 95e16; // 95%
+    uint256 private immutable _minSwapFeePercentage;
+
+    uint256 private _lastPauseCall;
+    uint256 private _lastRebalanceCall;
+
+    event PoolRebalancing(uint256 indexed startTime, uint256 endTime);
+
+    constructor(IVault vault, uint256 minSwapFeePercentage) {
+        // Get poolId from the factory.
+        bytes32 poolId = IManagedPool(ILastCreatedPoolFactory(msg.sender).getLastCreatedPool()).getPoolId();
+
+        // Verify that this is a real Vault and the pool is registered - this call will revert if not.
+        vault.getPool(poolId);
+
+        // Set the global tokens variables
+        (IERC20[] memory tokens, , ) = vault.getPoolTokens(poolId);
```

Comment:
same redundant validation comment as the weight changer controller. @rabmarut opinion here on redundancy vs controller example uniformity?

Reply:
I'd remove the redundancy.